RAGchain/benchmark/dataset/search_qa.py

```diff
@@ -62,36 +58,48 @@ def __init__(self, run_pipeline: BaseRunPipeline,
         self.eval_size = evaluate_size
         self.run_pipeline = run_pipeline
 
-        # Delete duplicated question - answer - retrieval gt
-        self.dataset = self.dataset.groupby('question', as_index=False).agg(
-            {'context': lambda x: list(x), 'key': lambda x: list(x),
-             'answers': lambda x: list(x.tolist()), 'labels': lambda x: list(x)})
-
-        self.context = deepcopy(self.dataset[['key', 'context', 'labels']])
-
-        if evaluate_size is not None and len(self.dataset) > evaluate_size:
-            self.qa_data = self.dataset[:evaluate_size]
+        if evaluate_size is not None and len(self.qa_data) > evaluate_size:
+            self.qa_data = self.qa_data[:evaluate_size]
 
-    def ingest(self, retrievals: List[BaseRetrieval], db: BaseDB, ingest_size: Optional[int] = None):
+    def ingest(self, retrievals: List[BaseRetrieval], db: BaseDB, ingest_size: Optional[int] = None, random_state=None):
         """
         Ingest dataset to retrievals and db.
         :param retrievals: The retrievals that you want to ingest.
         :param db: The db that you want to ingest.
         :param ingest_size: The number of data to ingest. If None, ingest all data.
-        If you want to use context_recall metrics, you should ingest all data.
+        If the ingest size is excessively large, it results in prolonged processing times.
+        To address this, we shuffle the corpus and slice it according to the ingest size for testing purposes.
+        The reason for transforming the retrieval ground truth corpus into passages and ingesting it is to enable
+        retrieval to retrieve the retrieval ground truth within the database.
+        This dataset has many retrieval ground truths per query, so it is recommended to set the ingest size to a small value.
+        :param random_state: A random state to fix the shuffled corpus to ingest.
+        Types are like these. int, array-like, BitGenerator, np.random.RandomState, np.random.Generator, optional
         """
 
-        ingest_data = deepcopy(self.context)
+        ingest_data = deepcopy(self.corpus)
+        gt_ingestion = list(itertools.chain.from_iterable(deepcopy([gt for gt in self.qa_data['retrieval_gt']])))
+
+        # Retrieval ground truth ingestion
+        gt_df = ingest_data[ingest_data['doc_id'].isin(gt_ingestion)]
+        gt_passages = gt_df.apply(self.__make_passages, axis=1).tolist()
+
         if ingest_size is not None:
             # ingest size must be larger than evaluate size.
             if ingest_size >= self.eval_size:
```

Comment:
if ingest_size is not None but self.eval_size is None, it occurs error. 
Fix this pls.